tcms/rpc/tests/test_testexecution.py

```diff
@@ -622,6 +624,31 @@ def test_update_with_no_perm(self):
                 self.execution_1.pk, {"stop_date": timezone.now()}
             )
 
+    def test_create_actual_duration(self):
+        test_execution = TestExecution()
+
+        test_execution.start_date = datetime.strptime("2020-05-05 09:00:00", "%Y-%m-%d %H:%M:%S")
+        test_execution.stop_date = datetime.strptime("2020-05-06 09:00:00", "%Y-%m-%d %H:%M:%S")
+
+        actual_duration_result = test_execution.stop_date - test_execution.start_date
+        self.assertEqual(test_execution.actual_duration,actual_duration_result)
+
+    def test_update_actual_duration(self):
+        test_execution = TestExecution()
+        execution = TestExecutionFactory(tested_by=None)
+
+        result = self.rpc_client.TestExecution.update(
+            execution.pk,
+            {
+                "start_date": "2021-02-25 12:12:12",
+                "stop_date": "2021-02-28 12:12:12",
+            },
+        )
+
+        execution.refresh_from_db()
+
+        actual_duration_result = result["stop_date"] - result["start_date"]
+        self.assertEqual(test_execution.actual_duration, actual_duration_result)
```

Comment:
What is the reasoning behind this test scenario compared to the previous one ?

Reply:
At first I was thinking this test can update either the start_date or the stop_date fields again with static data and check if the result from the actual_duration property has also been updated. I know there is already a method which checks when start_date and stop_date are crashing if they’re updated: 
`        self.assertEqual(result["start_date"], execution.start_date)
        self.assertEqual(result["stop_date"], execution.stop_date)`
but in the case that I wrote, I was trying to check if actual_duration property would update when either the stop_date or start_date changed and see its expected vs. actual result.

 But now Im thinking this case is not needed since in the previous test, I also gave static data and checked if the actual_duration property works as expected.

Reply:
So not only the 2 scenarios are checking exactly the same thing, but also `actual_duration` is a property, aka dynamically calculated field of the Class/object. As long as the mathematics behind it are correct it doesn't matter how often or which of the underlying fields are updated not does it matter how they are updated (API, UI, ORM).